butane_cli/src/lib.rs

```diff
@@ -263,10 +264,63 @@ pub fn get_migrations(base_dir: &Path) -> Result<FsMigrations> {
     Ok(migrations::from_root(root))
 }
 
+/// Extract the directory of a cargo workspace member identified by PackageId
+pub fn extract_package_directory(
+    packages: &[cargo_metadata::Package],
+    package_id: cargo_metadata::PackageId,
+) -> Result<std::path::PathBuf> {
+    let pkg = packages
+        .iter()
+        .find(|p| p.id == package_id)
+        .ok_or(anyhow::anyhow!("No package found"))?;
+    // Strip 'Cargo.toml' from the manifest_path
+    let parent = pkg.manifest_path.parent().unwrap();
+    Ok(parent.to_owned().into())
+}
+
+/// Find all of the `.butane` directories in the cargo workspace
+pub fn find_butane_dirs_in_workspace() -> Result<Vec<PathBuf>> {
+    let metadata = MetadataCommand::new().no_deps().exec()?;
+    let workspace_members = metadata.workspace_members;
+
+    let mut possible_directories: Vec<PathBuf> = vec![];
+    // Find the first workspace member with a .butane
+    for member in workspace_members {
+        let package_dir = extract_package_directory(&metadata.packages, member)?;
+        let member_butane_dir = package_dir.join(".butane/");
+
+        if member_butane_dir.exists() {
+            possible_directories.push(member_butane_dir);
+        }
+    }
+    Ok(possible_directories)
+}
+
+/// Find one `.butane` directory if only one exists in the cargo workspace
+pub fn find_butane_dir_in_workspace() -> Result<PathBuf> {
+    let possible_directories = find_butane_dirs_in_workspace()?;
+
+    match possible_directories.len() {
+        0 => Err(anyhow::anyhow!("No .butane exists")),
+        1 => Ok(possible_directories[0].to_owned()),
+        _ => Err(anyhow::anyhow!("Multiple .butane exists")),
+    }
+}
+
+/// Find a .butane directory to act as the base for butane.
 pub fn base_dir() -> Result<PathBuf> {
-    std::env::current_dir()
-        .map(|d| d.join(".butane"))
-        .map_err(|e| e.into())
+    let current_directory = std::env::current_dir()?;
+    let local_butane_dir = current_directory.join(".butane/");
+
+    if !local_butane_dir.exists() {
+        if let Ok(member_butane_dir) = find_butane_dir_in_workspace() {
```

Comment:
errors in related to running outside of a cargo project are now ignored here.  They are silently ignored, and that could be improved, but I would like to add `-p`/`--path` in a separate PR, and then the error messages can be improved to tell the user to use those flags.